hawkular-alerters/hawkular-elasticsearch-alerter/hawkular-elasticsearch-alerter-lib/src/main/java/org/hawkular/alerter/elasticsearch/ElasticSearchAlerter.java

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright 2015-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerter.elasticsearch;
+
+import static org.hawkular.alerter.elasticsearch.ServiceNames.Service.ALERTS_SERVICE;
+import static org.hawkular.alerter.elasticsearch.ServiceNames.Service.DEFINITIONS_SERVICE;
+import static org.hawkular.alerter.elasticsearch.ServiceNames.Service.PROPERTIES_SERVICE;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.Resource;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.hawkular.alerts.api.model.trigger.Trigger;
+import org.hawkular.alerts.api.services.AlertsService;
+import org.hawkular.alerts.api.services.DefinitionsService;
+import org.hawkular.alerts.api.services.DistributedEvent;
+import org.hawkular.alerts.api.services.PropertiesService;
+import org.jboss.logging.Logger;
+
+/**
+ * This is the main class of the ElasticSearch Alerter.
+ *
+ * The ElasticSearch Alerter will listen for triggers tagged with "ElasticSearch" tag. The Alerter will schedule a
+ * periodically query to an ElasticSearch system with the info provided from the tagged trigger context. The Alerter
+ * will convert ElasticSearch documents into Hawkular Alerting Events and send them into the Alerting engine.
+ *
+ * The ElasticSearch Alerter uses the following conventions for trigger tags and context:
+ *
+ * <pre>
+ *
+ * - [Required]    trigger.tags["ElasticSearch"] = "<timestamp field>"
+ *
+ *   Documents fetched from ElasticSearch needs a date field to indicate the timestamp.
+ *   This timestamp will be used in the queries to fetch documents in interval basis.
+ *
+ *   Timestamps are expected to follow patterns:
+ *
+ *      "yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ"
+ *      "yyyy-MM-dd'T'HH:mm:ssZ"
+ *
+ *   i.e.   trigger.tags["ElasticSearch"] = "@timestamp"
```

Comment:
yes, exactly, it identifies and groups the required alerter context from domain-specific context.  You don't see any value in having that convention?